### ISSUE TITLE
pkgconfig file should be relative to DESTDIR

### ISSUE THREAD

**Issue by chrisfosterelli** [lz4/lz4#456]
When providing a `$DESTDIR` and installing with `make install`, the resulting `liblz4.pc` file that is stored in `$DESTDIR/usr/local/lib/pkgconfig/` assumes the package was installed on the root and does not include the `$DESTDIR`.

This means that, when providing `PKG_CONFIG_PATH=$DESTDIR/usr/local/lib/pkgconfig/` to a project that depends on lz4, the build fails since it can't find the includes or library and the file has to be manually edited or the parent project has to be compiled with library/include flags (which defeats the purpose of pkgconfig) to get a build.

Other projects, like curl, seem to generate the pkgconfig file relative to the staged install directory so I thought I'd open an issue!

**Comment by Cyan4973** [comment]
Thanks for notification @chrisfosterelli .
Yes, this should be fixed

**Comment by felixhandte** [comment]
@chrisfosterelli, I believe the existing behavior is actually correct. I think you want to be setting `$PREFIX`, not `$DESTDIR`. `$DESTDIR` is intended to be used for staging an install through a temporary location; the files are still expected eventually to end up in `$PREFIX`.

See [the GNU convention](https://www.gnu.org/prep/standards/html_node/DESTDIR.html) for reference.

**Comment by chrisfosterelli** [comment]
@felixhandte I'm only very barely familiar with these things so I may be incorrect. I was attempting to setup a staged install in a temporary location, hence using `DESTDIR`.

I was cross compiling a project that utilizes lz4 as one of the dependencies. I was cross compiling the dependencies as staged installs, and then cross compiling the master project by passing references to the pkgconfig's for the dependencies. lz4 was the only project this failed on, and when I looked at the pkgconfig it was because the paths did not include the `DESTDIR` to the staged location I was storing it in (while the other projects did).

Should this have been done with `PREFIX`? I did not want to install these cross compiled dependencies into the host.

**Comment by terrelln** [comment]
I checked curl and zlib, both don't include `DESTDIR` in their `pkg-config` files:

```
git clone https://github.com/curl/curl
cd curl
mkdir build install
cd build
cmake ..
make -j
make install DESTDIR=../install
less ../install/usr/local/lib/pkgconfig/libcurl.pc
# Doesn't include ../install
```

```
git clone https://github.com/madler/zlib
./configure
mkdir install
make -j
make install DESDIR=install
less install/usr/local/lib/pkgconfig/zlib.pc
# Doesn'nt include install
```

**Comment by felixhandte** [comment]
@chrisfosterelli, thanks for the context! Yes, I believe you should still be setting `$PREFIX`, not `$DESTDIR`, since you intend to use the artifacts in that location. By overriding the prefix (to your destdir plus the default prefix value, `/usr/local`, e.g. `/home/felixh/staging/usr/local`), you will avoid touching your system installs.

**Comment by terrelln** [comment]
@chrisfosterelli Build/installation software **should** support building and installing to a staging directory, but it doesn't always work in practice, or is more complicated than it should be. I'm not sure what is breaking for you, but you might consider building in a chroot to avoid the problem altogether.

**Comment by chrisfosterelli** [comment]
Ahhh, okay. The other projects were definitely relative, which is interesting. I think you're right @felixhandte that I should use prefix, I may have set it on the other projects and missed it here, and then misinterpreted that to think this project needed a `DESTDIR`. I was in a "hack to get things working" mode so I'll revisit this and figure out where I went wrong haha.

At any rate, @terrelln showed that `DESTDIR` is not what I did to get the behaviour I had so I apologize for the misdirection and thank you all for taking the time to explain 😄
